src/utils/screenshotsaver.cpp

```diff
@@ -99,6 +99,55 @@ bool ScreenshotSaver::saveToFilesystem(const QPixmap& capture,
     return ok;
 }
 
+QString ScreenshotSaver::ShowSaveFileDialog(QWidget* parent,
+                                            const QString& title,
+                                            const QString& directory,
+                                            const QString& filter)
+{
+#if defined(Q_WS_WIN) || defined(Q_WS_MAC)
+    return QFileDialog::getSaveFileName(parent, title, directory, filter);
+#else
+    QFileDialog dialog(parent, title, directory, filter);
+    if (parent) {
+        dialog.setWindowModality(Qt::WindowModal);
+    }
+
+    QRegExp filter_regex(QLatin1String("(?:^\\*\\.(?!.*\\()|\\(\\*\\.)(\\w+)"));
+    QStringList filters = filter.split(QLatin1String(";;"));
+
+    dialog.setAcceptMode(QFileDialog::AcceptSave);
+
+    dialog.selectNameFilter(ConfigHandler().getSaveAsFileExtension());
+    if (dialog.exec() == QDialog::Accepted) {
+
+        ConfigHandler().setSaveAsFileExtension(dialog.selectedNameFilter());
+        QString file_name = dialog.selectedFiles().first();
+        QFileInfo info(file_name);
+
+        if ((dialog.selectedNameFilter() == "By extension [default: *.png]")) {
```

Comment:
"By extension [default: *.png]" should be defined as a variable so when it is re-used on line 174 there is no chance of it only changing in once place causing a future bug.

Reply:
Yeah, good point. I've moved it to const variables